src/engine/structures.rs

```diff
@@ -178,6 +188,14 @@ impl<T: HasName + HasUuid> Table<T> {
             (Some(name_item), Some(uuid_item)) => vec![name_item, uuid_item],
         }
     }
+
```

Comment:
My reading of ```Programming Rust``` was that by implementing ```IntoIterator``` in various ways that methods like these would be unnecessary. Could you explain more?